core/lib/zksync_core/src/lib.rs

```diff
@@ -630,6 +631,26 @@ pub async fn initialize_components(
                     Arc::new(ValidiumModeL1BatchCommitDataGenerator {})
                 }
             };
+
+        let current_commitment_mode_eth_response = eth_client
+            .call(
+                CallRequest::builder()
+                    .to((&contracts_config).executor_facet_addr.clone())
+                    .data(serde_json::from_str::<Bytes>(r#""0x252e5966""#).unwrap()) // The selector of "getPubdataPriceMode()"
+                    .build(),
+            )
+            .await
+            .unwrap();
+
+        let current_commitment_mode = L1BatchCommitDataGeneratorMode::from_eth_response(
+            &current_commitment_mode_eth_response,
+        )
+        .unwrap();
+
+        if current_commitment_mode != state_keeper_config.l1_batch_commit_data_generator_mode {
+            panic!("The selected L1BatchCommitDataGeneratorMode does not match the existing commitment mode");
+        }
```

Comment:
I'm not sure it would be better, but note this is equivalent to just using `assert_eq`.